AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/AWSAuthConfirmSignUpTask.swift

```diff
@@ -41,7 +41,7 @@ class AWSAuthConfirmSignUpTask: AuthConfirmSignUpTask, DefaultLogger {
                                            environment: userPoolEnvironment)
             _ = try await client.confirmSignUp(input: input)
             log.verbose("Received success")
-            return AuthSignUpResult(.done)
+            return AuthSignUpResult(.done, userID: nil)
```

Comment:
Do you really need this change? `userID` is set to `nil` by default in the `init`.

Reply:
We don't *need* it. But I believe it provides some value over omitting it.

I would've preferred to have `AuthSignUpResult.init` without a default argument for `userID`, but doing so would be a breaking change*. Explicitly passing the `nil` here provides some added context: "_we know there's no `userID` available when creating this_" vs. "_we may know that, or we may have missed that the argument exists_."

If explicitly passing `nil` has a drawback I'm not seeing (other than verbosity), I'm happy to reconsider.

*_we could add an overload with different labels or use `@_disfavoredOverload`, but they'd essentially be the same thing._

Reply:
Makes sense.. Thanks for explaining.